DotNet/Submission/Listeners/SubmitReportListener.cs

```diff
@@ -406,6 +417,83 @@ await File.WriteAllTextAsync(submissionDirectory + "/" + fileName, contents,
             }
         }
 
+        protected void GenerateSubmissionMetrics(Bundle? otherResourcesBundle, List<string>? patientFilesWritten, string facilityId, DateTime startDate, DateTime endDate )
+        {
+            if(otherResourcesBundle == null) { return; }    
+            if (patientFilesWritten == null || patientFilesWritten.Count == 0) { return; }
+
+            var totalResources = otherResourcesBundle.GetResources().Count();
+            var resourceTypes = new Dictionary<string, int>();
+            var medicationCodes = new Dictionary<string, int>();
+            foreach (var patientFile in patientFilesWritten)
```

Comment:
Fine for now... I wonder if we should consider threading this (in batches) in the future so that it processes ~1500 patients quicker.